src/main/scala/me/ex0ns/inlinexkcd/parser/XKCDHttpParser.scala

```diff
@@ -23,17 +23,19 @@ class XKCDHttpParser {
     * @param id the ID of the strip to fetch
     * @return a future that may contain the HttpResponse (if successful)
     */
-  def parseID(id: Int) : Future[_] = {
+  def parseID(id: Int): Future[_] = {
     val document = Comics.exists(id)
     document.flatMap {
       case true =>
         logger.debug(s"Document with id: $id already exists")
         Future.successful(true) // we do not want to stop at the first item we have in the DB
       case false =>
-        HttpRequest(s"http://xkcd.com/$id/info.0.json").send().map(httpResponse => {
-          Comics.insert(httpResponse.body)
-          httpResponse
-        })
+        HttpRequest(s"http://xkcd.com/$id/info.0.json")
+          .send()
+          .map(httpResponse => {
+            Comics.insert(httpResponse.body)
+            httpResponse
```

Comment:
There seems to be an concurrent issue (the bot send the notification many times to the same group), problem could be here, wait for the result of the insert